src/particle_filter.cpp

```diff
@@ -1,33 +1,95 @@
 #include "particle_filter.h"
 
 #include <algorithm>
+#include <cmath>
+
+#include <ros/console.h>
 
 // ----------------------------------------------------------------------------------------------------
 
-ParticleFilter::ParticleFilter() : i_current_(0)
+ParticleFilter::ParticleFilter() : min_samples_(0), max_samples_(0), kld_err_(0), kld_z_(0), alpha_slow_(0),
+    alpha_fast_(0), w_slow_(0), w_fast_(0), i_current_(0), kd_tree_(nullptr)
 {
 }
 
 // ----------------------------------------------------------------------------------------------------
 
+void ParticleFilter::configure(tue::Configuration config)
+{
+    // Extra variable needed as tue::Configuration doesn't support
+    // unisgned interters
```

Comment:
So does that mean you're implicitly casting the signed int to unsigned ints? In that case it might be better to make it explicit (doesn't the compiler give warnings?)

Reply:
No warnings. It is not possible to add unsigned integers in the current implementation of `tue_config`.  Or do you mean adding `static_cast`?